src/main/java/school/hei/haapi/endpoint/event/EventProducer.java

```diff
@@ -108,4 +114,12 @@ private void checkResponse(PutEventsResponse response) {
       log.info("Following events were successfully sent: {}", successfulEntries);
     }
   }
+
+  public boolean isPayloadValid(List<TypedEvent> events) {
```

Comment:
`isXxx` is not supposed to throw an exception. It should only return either `true` or `false`.

If you want to do it through an exception thrown from `EventProducer` (which is something I like), use a `private void checkPayload(List)` instead. It will throw BadRequest in case of invalidity, and nothing in case of validity. Also, as the message will be ultimately shown to consumer, use a more generic error message (as consumer does not need to know that we are using events behind). Eg something like `Request entries must be <= ...`